eve/main.yml

```diff
@@ -0,0 +1,48 @@
+---
+version: 0.2
+
+branches:
+  default:
+    stage: pre-merge
+
+models:
+  - Git: &clone
+      name: fetch source
+      repourl: '%(prop:git_reference)s'
+      shallow: True
+      retryFetch: True
+      haltOnFailure: True
+
+stages:
+  pre-merge:
+    worker: &master-worker
+      type: docker
+      path: eve/workers/master
+      volumes:
+        - '/home/eve/workspace'
+    steps:
+      - Git: *clone
+      - ShellCommand:
+          name: install
+          command: npm install
+      - ShellCommand:
+          name: run lint yml
+          command: npm run --silent lint_yml
+      - ShellCommand:
+          name: run lint
+          command: npm run --silent lint -- --max-warnings 0
+      - ShellCommand:
+          name: run lint_md
+          command: npm run --silent lint_md
+      - ShellCommand:
+          name: run test
+          command: npm run --silent test
+      - ShellCommand:
+          name: run ft_test
+          command: npm run ft_test
+      - ShellCommand:
+          name: run executables tests
+          command: |
+            cd lib/executables/pensieveCreds &&
+            ln -s ../../../node_modules . &&
+            npm install && npm test
```

Comment:
putting the package file in the worker dir, then link it at root, may give better code?

Reply:
In this case, it's another `package.json` file that is called. I've linked the main one which is located on the root directory to the worker. Not sure if will do the same with this one.

Reply:
and btw, this step takes exactly 4 seconds.